analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -41,14 +43,35 @@ private class TestSetup
         {
             public string Path { get; set; }
             public SonarDiagnosticAnalyzer Analyzer { get; set; }
+            public IEnumerable<MetadataReference> AdditionalReferences { get; set; } = new MetadataReference[] { };
         }
 
+        // Various classes that invoke all the `ReportDiagnosticWhenActive` methods in AnalysisContextExtensions
+        // We mention in comments the type of Context that is used to invoke (directly or indirectly) the `ReportDiagnosticWhenActive` method
         private readonly List<TestSetup> testCases = new List<TestSetup>(new[]
         {
+            // SyntaxNodeAnalysisContext
+            // S3244 - MAIN and TEST
             new TestSetup { Path = @"TestCases\AnonymousDelegateEventUnsubscribe.cs", Analyzer = new AnonymousDelegateEventUnsubscribe() },
+            // S2699 - TEST only
+            new TestSetup { Path = @"TestCases\TestMethodShouldContainAssertion.MsTest.cs", Analyzer = new TestMethodShouldContainAssertion(), AdditionalReferences = TestMethodShouldContainAssertionTest.GetMsTestReferences(Constants.NuGetLatestVersion)},
```

Comment:
Explicit constructor in `TestSetup` class can remove a lot of repetitive pain with named arguments.